src/xds/client.rs

```diff
@@ -141,7 +144,7 @@ impl<T: 'static + prost::Message + Default> RawHandler for HandlerWrapper<T> {
                 state.add_resource(key.type_url, key.name);
                 r
             })
-            .map(|raw| decode_proto::<T>(raw).unwrap())
+            .map(|raw| decode_proto::<T>(&raw).unwrap())
```

Comment:
why this change?  I think this means we do a clone() on the full proto which we shouldn't need?

Reply:
we don't need this changed now, i left there by mistake from another change I'm working on. s

It should matter in terms of `clone()`ing though. You can see that `decode_proto` doesn't do any additional clone() operations

If you prefer i can revert it from here and add it in the next PR I'm working on.

Reply:
sorry, `decode_proto` does clone the name now; i'll revert it from here and add it back in context so it will make more sense

Reply:
nvm, misread it

Reply:
I put hold incase you want to fix, feel free to remove